### PR TITLE
Implement the and query operator

### PR DESCRIPTION
**Pull request by bas080**

### CODE REVIEW

File: browser/search.mjs [hunk 1]
```diff
@@ -28,7 +28,18 @@ const inputRef = createRef()
 const root = document.getElementById('issue-search')
```
Comment (bas080) [inline browser/search.mjs:28]: Just a bit of cleanup.

File: browser/index.mjs [hunk 2]
```diff
@@ -54,6 +47,34 @@ const tokenIssues = (token) => {
...
+
+  if (token === 'or') {
+    const [next, ...nextRest] = rest
+
+    return issueMatches(nextRest, or(predicate, tokenMatches(next)))
+  }
+
+  return issueMatches(rest, and(predicate, tokenMatches(token)))
+}
+
```
Comment (bas080) [inline browser/index.mjs:77]: Rewrote to a recursive function. Would be nice to pull to a module and write tests for. Should do later.

File: browser/index.mjs [hunk 3]
```diff
@@ -54,6 +47,34 @@ const tokenIssues = (token) => {
   )
 }
 
+const alwaysTrue = always(true)
+const or = (before, token) => (issue) => before(issue) || token(issue)
+const and = (before, token) => (issue) => before(issue) && token(issue)
+const tokenMatches = (token) => {
+  token = token.startsWith('/') ? `${token.slice(1)}/` : token
```
Comment (antoniskalou) [inline browser/index.mjs:54]: is this for converting to a relative directory?

File: browser/index.mjs (hunk 2 above)
Comment (antoniskalou) [reply browser/index.mjs:77]: Very nice

File: browser/search.mjs [hunk 4]
```diff
@@ -28,7 +28,17 @@ const inputRef = createRef()
...
 root.innerHTML = ''
 
-const search = (state, push) => {
+const tokenIcon = byKey(
+  {
+    '#': '🏷️',
+    '@': '🧑',
+    '"': '🔤',
```
Comment (antoniskalou) [inline browser/search.mjs:35]: what's the quote for?

File: browser/search.mjs [hunk 5]
```diff
@@ -103,24 +113,26 @@ const search = (state, push) => {
   })
 
   const searchTokenItem = (token, index, tokens) => {
-    console.log(token, index, tokens)
     const count = issuesPerToken[token] || 0
 
     if (token === 'or') {
-      return html`<li class="issue-search-query-or-item">or</li>`
+      return html`<li class="">or</li>`
```
Comment (antoniskalou) [inline browser/search.mjs:119]: is the lack of class here intentional? why not `<li>or</li>` in that case?

File: browser/search.mjs [hunk 6]
```diff
@@ -103,24 +113,26 @@ const search = (state, push) => {
...
-        class="issue-search-query-item"
-        title="Remove ${token}"
-        value="${token}"
-        @click="${partial(onTokenRemove, index)}"
-      >
-        ${tokenIcon(token[0])} ${quoteOnWhitespace(token)}
-        <span class="badge badge-primary">${count}</span>
-      </button>
-    </li>`
+    return [
```
Comment (antoniskalou) [inline browser/search.mjs:122]: Why return a list here if its only ever containing one element?

File: browser/index.mjs (hunk 3 above)
Comment (bas080) [reply browser/index.mjs:54]: The query language uses `/<some-dir` but we want to match on `<some-dir>/` in the issue articles text.

Example

2023-12-19-190946_179x61_scrot

You can see that the text that represents this path does not start with a `/`, but if I want to match on it I do `/active`. So a quick and dirty trick is to change the search token `/active` to `active/`. It also happens to exclude filepaths (which used to also match).

File: browser/search.mjs (hunk 4 above)
Comment (bas080) [reply browser/search.mjs:35]: The query language supports `"` for tokens with spaces. This allows configuring an icon for the token button. 

2023-12-19-191411_424x203_scrot

File: browser/search.mjs (hunk 5 above)
Comment (bas080) [reply browser/search.mjs:119]: The class is no longer needed because it needs no styling. I'll remove the class.

File: browser/search.mjs (hunk 6 above)
Comment (bas080) [reply browser/search.mjs:122]: This array will contain two elements if the previous token was not an "or".

File: browser/search.mjs (hunk 4 above)
Comment (antoniskalou) [reply browser/search.mjs:35]: its a bit confusing to have an opening but no closing bracket

**Review by antoniskalou — APPROVED** [review]

**Pull request merged**